src/internal/connector/graph/errors.go

```diff
@@ -169,23 +167,9 @@ func IsMalware(err error) bool {
 }
 
 func IsMalwareResp(ctx context.Context, resp *http.Response) bool {
-	if resp == nil {
-		return false
-	}
-
 	// https://learn.microsoft.com/en-us/openspecs/sharepoint_protocols/ms-wsshp/ba4ee7a8-704c-4e9c-ab14-fa44c574bdf4
 	// https://learn.microsoft.com/en-us/openspecs/sharepoint_protocols/ms-wdvmoduu/6fa6d4a9-ac18-4cd7-b696-8a3b14a98291
-	if len(resp.Header) > 0 && resp.Header.Get("X-Virus-Infected") == "true" {
-		return true
-	}
-
-	respDump, err := httputil.DumpResponse(resp, true)
-	if err != nil {
-		logger.Ctx(ctx).Errorw("dumping http response", "error", err)
-		return false
-	}
-
-	if strings.Contains(string(respDump), string(malwareDetected)) {
+	if resp != nil && len(resp.Header) > 0 && resp.Header.Get("X-Virus-Infected") == "true" {
```

Comment:
`if foo { return true } return false` is redundant, and can be reduced to the conditional check.
```suggestion
	return resp != nil &&
	len(resp.Header) > 0 &&
	resp.Header.Get("X-Virus-Infected") == "true"
```